src/lib/github/prComments.ts

```diff
@@ -107,35 +107,35 @@ export const createPrComment = async (
   });
 
   let ghResponse;
-  let firstComment = false;
-  let commentUrl = '';
+  let firstComment = true;
+  let commentUrl = `/repos/${ghDetails.orgName}/${ghDetails.repoName}/issues/${ghDetails.prNumber}/comments`;
 
   ghResponse = await ghClient.get(
-    `/repos/${ghDetails.orgName}/${ghDetails.repoName}/issues/${ghDetails.prNumber}/comments`
+    commentUrl
   );
 
-  if (ghResponse.data.length === 0) {
-    firstComment = true
-  } else {
+  if (ghResponse.data.length != 0) {
     ghResponse.data.map((comments: any) => {
       if (comments.body.includes('******* Vulnerabilities report for commit')) 
       {
-        commentUrl = comments.url
+        if(!keepHistory)
```

Comment:
I'd move the keep history to the check of the data length above. No need to iterate over the loop for nothing. Wdyt?